Sources/SwiftDocC/SwiftDocCMarkdownOutput/MarkdownOutputNode.swift

```diff
@@ -0,0 +1,199 @@
+    public struct Metadata: Codable, Sendable {
+    
+        static let version = "0.1.0"
+        
+        public enum DocumentType: String, Codable, Sendable {
+            case article, tutorial, symbol
+        }
+        
+        public struct Availability: Codable, Equatable, Sendable {
+            
+            public let platform: String
+            public let introduced: String?
+            public let deprecated: String?
+            public let unavailable: Bool
+                        
+            public init(platform: String, introduced: String? = nil, deprecated: String? = nil, unavailable: Bool) {
+                self.platform = platform
+                // Can't have deprecated without an introduced
+                self.introduced = introduced ?? deprecated
+                self.deprecated = deprecated
+                // If no introduced, we are unavailable
+                self.unavailable = unavailable || introduced == nil
+            }
+            
+            // For a compact representation on-disk and for human and machine readers, availability is stored as a single string:
+            // platform: introduced -               (not deprecated)
+            // platform: introduced - deprecated    (deprecated)
+            // platform: -                          (unavailable)
+            public func encode(to encoder: any Encoder) throws {
+                var container = encoder.singleValueContainer()
+                try container.encode(stringRepresentation)
+            }
+            
+            public init(from decoder: any Decoder) throws {
+                let container = try decoder.singleValueContainer()
+                let stringRepresentation = try container.decode(String.self)
+                self.init(stringRepresentation: stringRepresentation)
+            }
+            
+            public var stringRepresentation: String {
+                var stringRepresentation = "\(platform): "
+                if unavailable {
+                    stringRepresentation += "-"
+                } else {
+                    if let introduced, introduced.isEmpty == false {
+                        stringRepresentation += "\(introduced) -"
+                        if let deprecated, deprecated.isEmpty == false {
+                            stringRepresentation += " \(deprecated)"
+                        }
+                    } else {
+                        stringRepresentation += "-"
+                    }
+                }
+                return stringRepresentation
+            }
+            
+            public init(stringRepresentation: String) {
+                let words = stringRepresentation.split(separator: ":", maxSplits: 1)
+                if words.count != 2 {
+                    platform = stringRepresentation
+                    unavailable = true
+                    introduced = nil
+                    deprecated = nil
+                    return
+                }
```

Comment:
FYI: If you used a guard-statement here, the compiler would help guarantee that this code path returns (or throws or otherwise exist) so that it's not possible to have a bug where the code falls through and also runs the code below.